plugins/rts54hub/fu-rts54hub-rtd21xx-device.c

```diff
@@ -282,15 +356,78 @@ fu_rts54hub_rtd21xx_ensure_version_unlocked (FuRts54hubRtd21xxDevice *self,
 	return TRUE;
 }
 
+static gboolean
+fu_rts54hub_rtd21xx_ensure_version_background (FuRts54hubRtd21xxDevice *self,
+					     GError **error)
+{
+	guint8 buf_rep[7] = { 0x00 };
+	guint8 buf_req[] = { ISP_BG_CMD_GET_FW_INFO };
+	guint8 buf[] = { ISP_BG_CMD_FW_UPDATE_EXIT };
+	g_autofree gchar *version = NULL;
+
+	if (!fu_rts54hub_rtd21xx_device_i2c_write (self,
+						   UC_ISP_SLAVE_ADDR,
+						   UC_BACKGROUND_OPCODE,
+						   buf_req, sizeof(buf_req),
+						   error)) {
+		g_prefix_error (error, "failed to get version number: ");
+		return FALSE;
+	}
+
+	/* wait for device ready */
+	g_usleep (300000);
```

Comment:
would this be better done as a retry until it works rather than fixed timeout?

Reply:
After sending the command, it takes time to wait for interrupt reception and FW processing to complete. 
Removing delay and directly using retry will cause the interrupt to be triggered all the time, resulting in incorrect reading value

Reply:
thanks for clarification.